cgsmiles/graph_layout.py

```diff
@@ -0,0 +1,222 @@
+    top we take care that fife membered rings and triangles
+    have the correct angle sum.
+
+    Parameters
+    ----------
+    graph: nx.Graph
+
+    Returns
+    -------
+    tuple([[abc.hashable,abc.hashable,abc.hashable,int])
+        tuple with node keys corresponding to the angle
+        and the reference angle value
+    """
+    triplets = find_triplets(graph)
+    angles = []
+    for triplet in triplets:
+        if graph.degree(triplet[1]) in [2, 3]:
+            angles.append((triplet[0],
+                          triplet[1],
+                          triplet[2],
+                          120))
+    return angles
+
+def assign_bonds(graph, pos):
+    """
+    Iterate over all edges and compute the distance
+    between the edges defined by the `pos` coordinate
+    array.
+
+    Parameters
+    ----------
+    graph: nx.Graph
+    pos: np.ndarray((n,2))
+
+    Returns
+    -------
+    tuple([abc.hashable, abc.hashable, float])
+        tuple of node keys and distance
+    """
+    bonds = []
+    for edge in graph.edges:
+        dist = np.linalg.norm(pos[edge[0]]-pos[edge[1]])
+        bonds.append((edge[0], edge[1], dist))
+    return bonds
+
+def angle_between(v1, v2):
+    """
+    Compute angle between two 2D vectors.
+    """
+    cos_theta = np.dot(v1, v2) / (np.linalg.norm(v1) * np.linalg.norm(v2))
+    angle_rad = np.arccos(np.clip(cos_theta, -1.0, 1.0))
+    angle_deg = np.degrees(angle_rad)
+    return angle_deg
+
+
+def bond_potential(atom_pos, ref):
+    """
+    Compute energy associated to the bonds.
+    """
+    energy = 10000*(ref-np.linalg.norm(atom_pos[1]-atom_pos[0]))**2
+    return energy
+
+def angle_potential(atom_pos, ref):
+    """
+    Compute the energy associated with an
+    angular potential.
+    """
+    v1 = atom_pos[1] - atom_pos[0]
+    v2 = atom_pos[1] - atom_pos[2]
+    angle = angle_between(v1, v2)
+    energy = (ref-angle_between(v1, v2))**2
+    return energy
+
+def nonbonded_potential(positions):
+    """
+    Compute a simple repulsive nonbonded potential
+    between all nodes whos positions are described by
+    `positions` array.
+    """
+    dist_matrix = squareform(pdist(positions, 'euclidean'))
+    dist_matrix = dist_matrix[dist_matrix > 10**-6]
+    energy = np.sum(np.power(dist_matrix, -8))
+    return energy
+
+
+def optimize_geometry_2D(pos, interactions, inter_methods):
+    """
+    Given a position array of shape (n, 2) and a dict of
+    interactions, optimize the positions to minimize the
+    energy.
+
+    Parameters
+    ----------
+    pos: np.ndarray((n, 2))
+        the inital positions
+    interactions: dict[str, list(tuple)]
+        a dict of interactions with key type and list of
+        interaction tuples
+
+    Returns
+    -------
+    np.ndarray((n, 2)), float
+        the optimized positions and final energy
+    """
+    n_atoms = len(pos)
+    atom_to_idx = OrderedDict(zip(list(pos.keys()), range(0, n_atoms)))
```

Comment:
```suggestion
    atom_to_idx = OrderedDict(zip(pos.keys(), range(0, n_atoms)))
```
Not sure OrderedDict is not deprecated.
Also consider:
```suggestion
    atom_to_idx = {n: idx for idx, n in enumerate(pos.keys())}
```